From 74263f22d194e6a2e6f36dd3f96b532652ff1d7d Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Thu, 29 Sep 2016 11:39:57 +0800
Subject: [PATCH 823/827] fpga: zynqmp: fix compiling warnings

drivers/fpga/zynqmp-fpga.c: In function 'zynqmp_fpga_ops_write':
drivers/fpga/zynqmp-fpga.c:60:2: warning: passing argument 1 of '__flush_cache_user_range' makes integer from pointer without a cast
  __flush_cache_user_range(kbuf, kbuf + size);
  ^
In file included from drivers/fpga/zynqmp-fpga.c:15:0:
./arch/arm64/include/asm/cacheflush.h:76:13: note: expected 'long unsigned int' but argument is of type 'char *'
 extern long __flush_cache_user_range(unsigned long start, unsigned long end);
             ^
drivers/fpga/zynqmp-fpga.c:60:2: warning: passing argument 2 of '__flush_cache_user_range' makes integer from pointer without a cast
  __flush_cache_user_range(kbuf, kbuf + size);
  ^
In file included from drivers/fpga/zynqmp-fpga.c:15:0:
./arch/arm64/include/asm/cacheflush.h:76:13: note: expected 'long unsigned int' but argument is of type 'char *'
 extern long __flush_cache_user_range(unsigned long start, unsigned long end);

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/fpga/zynqmp-fpga.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/fpga/zynqmp-fpga.c b/drivers/fpga/zynqmp-fpga.c
index 5119cbf..be5f564 100644
--- a/drivers/fpga/zynqmp-fpga.c
+++ b/drivers/fpga/zynqmp-fpga.c
@@ -57,7 +57,8 @@ static int zynqmp_fpga_ops_write(struct fpga_manager *mgr,
 		return -ENOMEM;
 
 	memcpy(kbuf, buf, size);
-	__flush_cache_user_range(kbuf, kbuf + size);
+	__flush_cache_user_range((unsigned long)kbuf,
+				(unsigned long)kbuf + size);
 
 	/**
 	 * Translate size from bytes to number of 32bit words that
-- 
2.9.3

